### PR TITLE
Better Error Logging

### PR DESCRIPTION
**Pull request by lfparis**
* modified `raise_or_log_error() `to process from `ClientResponse`

### CODE REVIEW

File: airbase/airtable.py [hunk 1]
```diff
@@ -193,13 +217,7 @@ async def get_bases(self) -> Optional[List[Base]]:
                 self.logger.info(f"Fetched: {len(self.bases)} bases")
 
             else:
-                error_msg = self.get_error_message(
-                    method="get",
```
Comment (cancan101) [inline airbase/airtable.py:197]: do you care about losing any of this info (method / obj)?

File: airbase/airtable.py [hunk 2]
```diff
@@ -232,46 +250,39 @@ async def get_base(
...
+                    f"Fetched Base with {key if key else 'value'}: '{value}'"
+                )  # noqa: E501
                 return base
         else:
-            error_msg = self.get_error_message(
-                method="get",
-                obj='base',
-            )
-            self.raise_or_log_error(error_msg)
+            error_msg = f"ERROR: Base with {key if key else 'value'}:'{value}' not found"  # noqa: E501
```
Comment (cancan101) [inline airbase/airtable.py:258]: is this prefix needed? I assume the error logger handles that?

File: airbase/airtable.py [hunk 3]
```diff
@@ -93,33 +93,57 @@ async def _request(self, *args, **kwargs) -> Optional[ClientResponse]:
...
 
-    def raise_or_log_error(self, error_msg: str) -> None:
+    async def raise_or_log_error(
+        self,
+        res: Optional[ClientResponse],
+        error_msg: str = "",
+    ) -> None:
+        error_msg = await self.get_error_message(res) if res else error_msg
         if self.raise_for_status:
             raise AirbaseException(error_msg)
```
Comment (cancan101) [inline airbase/airtable.py:103]: can you also pass the res object to `AirbaseException` or perhaps subclass it to `AirbaseResponseException` in the case res is not None ?

File: airbase/airtable.py (hunk 1 above)
Comment (lfparis) [reply airbase/airtable.py:197]: The updated `get_error_message()` coroutine leverages the response to determine the method and object of the request

File: airbase/airtable.py (hunk 2 above)
Comment (lfparis) [reply airbase/airtable.py:258]: Corrext. Prefix is not needed.

File: airbase/airtable.py [hunk 4]
```diff
@@ -93,33 +93,57 @@ async def _request(self, *args, **kwargs) -> Optional[ClientResponse]:
             else:
                 return res
 
-    def raise_or_log_error(self, error_msg: str) -> None:
+    async def raise_or_log_error(
```
[the remark this answers is not in the document]
Comment (lfparis) [reply airbase/airtable.py:96]: see updated changes

File: airbase/airtable.py (hunk 3 above)
Comment (lfparis) [reply airbase/airtable.py]: See latest changes. Added subclass `AirbaseResponseException` and passed on ClientResponse

File: airbase/airtable.py [hunk 5]
```diff
@@ -677,12 +674,11 @@ async def get_records(
         while True:
             async with self.base.semaphore:
                 res = await self._request("get", self.url, params=params)
-            data = await self._get_data(res)
             if not self._is_success(res):
-                error_msg = f"{res.status}: Records for Table: {self.name} could not be retreived -> {data.get('error')}"  # noqa: E501
-                self.raise_or_log_error(error_msg)
+                # error_msg = f"{res.status}: Records for Table: {self.name} could not be retreived -> {data.get('error')}"  # noqa: E501
```
Comment (cancan101) [inline airbase/airtable.py:678]: kill dead code?

File: airbase/airtable.py [hunk 6]
```diff
@@ -232,46 +255,39 @@ async def get_base(
                 if bases:
                     base = bases[0]
             if base:
-                self.logger.info(f"Fetched Base with {key if key else 'value'}: {value}")  # noqa: E501
+                self.logger.info(
+                    f"Fetched Base with {key if key else 'value'}: '{value}'"
+                )  # noqa: E501
```
Comment (cancan101) [inline airbase/airtable.py:260]: is the noqa still needed here?

File: airbase/airtable.py (hunk 5 above)
Comment (lfparis) [reply airbase/airtable.py]: removed

**Pull request merged**